compiler/rustc_typeck/src/check/demand.rs

```diff
@@ -1173,6 +1173,16 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 }
                 true
             }
+            (&ty::Int(ty::IntTy::I32) | &ty::Uint(ty::UintTy::U32), &ty::Char)
+            | (&ty::Char, &ty::Int(ty::IntTy::I32) | &ty::Uint(ty::UintTy::U32)) => {
```

Comment:
May want to match on all sorts of integers here, not just `i/u32`. E.g. `char as u64` is also fine.